x-pack/legacy/plugins/siem/public/components/timeline/insert_timeline_popover/use_insert_timeline.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useState } from 'react';
+import { useBasePath } from '../../../lib/kibana';
+import { CursorPosition } from '../../markdown_editor';
+import { FormData, FormHook } from '../../../shared_imports';
+
+export const useInsertTimeline = <T extends FormData>(form: FormHook<T>, fieldName: string) => {
+  const basePath = useBasePath();
```

Comment:
@XavierM this just is not working.... left it broken for helps

Reply:
I will update useBaePath to use `KibanaServices.get().http.basePath`

Reply:
Alright my bad here, I think I got confused like you. you just need to do that to get the shareable URL ` const basePath = window.location.origin + useBasePath();` and you can keep the useBasePath the way it is.